### PR TITLE
Remove V2

### PR DESCRIPTION
**Pull request by alecholmez**
@mattklein123 @snowp this should be good now, had to fix some conflicts from the original PR going in

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ build:
 	@go build ./pkg/... ./envoy/...
 
 # TODO(mattklein123): See the note in TestLinearConcurrentSetWatch() for why we set -parallel here
-# This should be removed.
+# This should be
```
Comment (snowp) [inline Makefile:16]: Revert this change

**Review by snowp — CHANGES_REQUESTED** [review]
Just one nit, otherwise LGTM!

File: Makefile (hunk 1 above)
Comment (alecholmez) [reply Makefile:16]: Oops! Reverting

Comment (alecholmez) [reply Makefile]: @snowp all good!

**Comment by mattklein123** [comment]
@alecholmez I'm fine merging this, but I think now you should be able to fix this TODO: https://github.com/envoyproxy/go-control-plane/blob/main/Dockerfile.ci#L3

I think switching to envoy-dev:latest should "just work" if you want to give it a shot?

**Comment by alecholmez** [comment]
@mattklein123 okay sweet I'll change that and see how it goes

**Comment by alecholmez** [comment]
@mattklein123 hmmmm, seems like the integration tests are failing now. `envoy:v1.14.1` passes but `envoy-dev:latest` fails with this:

```
env XDS=xds build/integration.sh
Envoy log: envoy.xds.log
2021/04/07 19:31:17 upstream listening HTTP/1.1 on 18080
2021/04/07 19:31:17 waiting for the first request...
2021/04/07 19:31:17 access log server listening on 18090
2021/04/07 19:31:17 gateway listening HTTP/1.1 on 18001
2021/04/07 19:31:17 management server listening on 18000
2021/04/07 19:31:17 stream 1 open for type.googleapis.com/envoy.service.runtime.v3.Runtime
2021/04/07 19:31:17 initial snapshot {Xds:xds Version: UpstreamPort:18080 BasePort:9000 NumClusters:4 NumHTTPListeners:2 NumTCPListeners:2 NumRuntimes:1 TLS:false}
2021/04/07 19:31:17 executing sequence updates=3 request=5
2021/04/07 19:31:17 update snapshot v0
2021/04/07 19:31:17 open watch 1 for type.googleapis.com/envoy.service.runtime.v3.Runtime[runtime-0] from nodeID "test-id", version ""
2021/04/07 19:31:17 respond open watch 1[runtime-0] with new version "v0"
2021/04/07 19:31:17 respond type.googleapis.com/envoy.service.runtime.v3.Runtime[runtime-0] version "" with version "v0"
2021/04/07 19:31:17 request batch 0, ok 0, failed 4, pass false
2021/04/07 19:31:17 stream 2 open for type.googleapis.com/envoy.config.cluster.v3.Cluster
2021/04/07 19:31:17 open watch 2 for type.googleapis.com/envoy.service.runtime.v3.Runtime[runtime-0] from nodeID "test-id", version "v0"
2021/04/07 19:31:17 respond type.googleapis.com/envoy.config.cluster.v3.Cluster[] version "" with version "v0"
2021/04/07 19:31:18 stream 3 open for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
2021/04/07 19:31:18 stream 4 open for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
2021/04/07 19:31:18 stream 5 open for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
2021/04/07 19:31:18 stream 6 open for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
2021/04/07 19:31:18 open watch 3 for type.googleapis.com/envoy.config.cluster.v3.Cluster[] from nodeID "test-id", version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-0] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-2] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-1] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-3] version "" with version "v0"
2021/04/07 19:31:18 open watch 4 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-0] from nodeID "test-id", version "v0"
2021/04/07 19:31:18 open watch 5 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-2] from nodeID "test-id", version "v0"
2021/04/07 19:31:18 stream 7 open for type.googleapis.com/envoy.config.listener.v3.Listener
2021/04/07 19:31:18 open watch 6 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-1] from nodeID "test-id", version "v0"
2021/04/07 19:31:18 open watch 7 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment[cluster-v0-3] from nodeID "test-id", version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 respond type.googleapis.com/envoy.config.listener.v3.Listener[] version "" with version "v0"
2021/04/07 19:31:18 request batch 1, ok 2, failed 2, pass false
```

I'll look through the scripts to see if there is anything that stands out to me

**Comment by alecholmez** [comment]
Ah seeing this:
```
[2021-04-07 19:42:26.618][8716][warning][config] [source/common/config/grpc_subscription_impl.cc:127] gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) listener-9001: V2 (and AUTO) xDS transport protocol versions are deprecated in log_name: "echo"
grpc_service {
  envoy_grpc {
    cluster_name: "xds_cluster"
  }
}
. The v2 xDS major version is deprecated and disabled by default. Support for v2 will be removed from Envoy at the start of Q1 2021. You may make use of v2 in Q4 2020 by following the advice in https://www.envoyproxy.io/docs/envoy/latest/faq/api/transition.
listener-9000: V2 (and AUTO) xDS transport protocol versions are deprecated in log_name: "echo"
grpc_service {
  envoy_grpc {
    cluster_name: "xds_cluster"
  }
}
```
Must be some remnants of the integration test. Will look into that

**Comment by alecholmez** [comment]
Okay should be all good now @mattklein123

**Comment by mattklein123** [comment]
@alecholmez if it's not easy to fix I would say let's just merge and do a follow up, but it seems like you might have already fixed it.

**Comment by alecholmez** [comment]
> @alecholmez if it's not easy to fix I would say let's just merge and do a follow up, but it seems like you might have already fixed it.

😁 haha yup fixed it as you wrote that. Just need to wait for the build

**Review by mattklein123 — DISMISSED** [review]
Fantastic thanks. Did you also grep for v2 in pkg to make sure there is nothing else like the last bug?

**Comment by alecholmez** [comment]
> Fantastic thanks. Did you also grep for v2 in pkg to make sure there is nothing else like the last bug?

Just did one last final grep and removed one more line. All good now.

**Review by snowp — APPROVED** [review]

**Review by mattklein123 — APPROVED** [review]
Thank you so much!

**Comment by alecholmez** [comment]
> Thank you so much!

Sure thing! Glad I can help 😁
